torch/ao/sparsity/_experimental/data_sparsifier/README.md

```diff
@@ -0,0 +1,128 @@
+# Data Sparsifier
+## Intro
+The data sparsifier inherits from the `BaseSparsifier` class. It attempts to sparsify data tensors in general (trainable and non-trainable).
+
+## Implementation Details
+The data sparsifier does not receive a model or a layer to sparsify. Hence, the mask needs to be owned by the data sparsifier. This is acheived by introducing a private container model that registers the data as a parametrized buffer.
+
+The BaseDataSparsifier handles all the housekeeping while allowing the user to just implement the `update_mask` logic in their implementation.
+
+## Supported data
+1. torch tensors (torch.Tensor)
+2. parameters (nn.Parameter)
+3. embedding and embedding bags (nn.Embeddings / nn.EmbeddingBag)
+
+## API details
+`BaseDataSparsifier`: base class with abstract method `update_mask` that computes the new mask for all the data.
+
+`add_data`: Accepts name, data tuple and registers the data as a parametrized buffer inside the container model. Note that the data is always associated to a name. A custom sparse config can be provided along with the name, data pair. If not provided, the default config will be applied while doing the sparsification.
+If the named data already exists, then it is replaced with the new data. The config and mask will be retained for the new data unless not specified to.
+To not the old mask, set `use_old_mask=False`. If the `config` is explicitly passed in, it will be updated.
+
+**Note**: name containing '.' is not a valid name for the data sparsifier
```

Comment:
nit: Paraphrase this, please: "Data sparsifier `name` argument cannot have '.' in it."